pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
 
-        using_valid_distributed = self.use_ddp or self.use_ddp2
-        if self.num_nodes > 1 and not using_valid_distributed:
-            # throw error to force user to choose a supported strategy type such as ddp or ddp2
-            raise MisconfigurationException(
-                "Your chosen strategy does not support `num_nodes > 1`. Please set `strategy=('ddp'|'ddp2')`."
-            )
+    @property
+    def tpu_cores(self) -> int:
+        return self.devices
 
-    def _set_horovod_backend(self):
-        self.check_horovod()
-        self._strategy_type = _StrategyType.HOROVOD
+    @property
+    def ipus(self) -> int:
+        return self.devices
 
-        # Initialize Horovod to get rank / size info
-        hvd.init()
-        if self.has_gpu:
-            # Horovod assigns one local GPU per process
-            self.parallel_device_ids = list(range(hvd.local_size()))
+    @property
+    def num_gpus(self) -> int:
+        if isinstance(self.accelerator, GPUAccelerator):
+            return self.devices
         else:
-            self.num_processes = hvd.local_size()
+            return 0
 
-    def check_interactive_compatibility(self):
-        """Raises a `MisconfigurationException` if the accelerator and/or plugin is not compatible with an
-        interactive environment."""
-        from pytorch_lightning.utilities import _IS_INTERACTIVE
+    # def parallel_device_ids():
+    @property
+    def gpus(self):
+        return self._gpus
+        # if isinstance(self.accelerator, GPUAccelerator) else 0
 
-        if _IS_INTERACTIVE and self._strategy_type is not None and not self._strategy_type.is_interactive_compatible():
-            raise MisconfigurationException(
-                f"`Trainer(strategy={self._strategy_type.value!r})` or"
-                f" `Trainer(accelerator={self._strategy_type.value!r})` is not compatible with an interactive"
-                " environment. Run your code as a script, or choose one of the compatible backends:"
-                f" {', '.join(_StrategyType.interactive_compatible_types())}."
-                " In case you are spawning processes yourself, make sure to include the Trainer"
-                " creation inside the worker function."
-            )
+    @property
+    def parallel_device_ids(self):
+        return [i for i in range(len(self.parallel_devices))] if isinstance(self.accelerator, GPUAccelerator) else None
 
-    def check_horovod(self):
-        """Raises a `MisconfigurationException` if the Trainer is not configured correctly for Horovod."""
-        if not _HOROVOD_AVAILABLE:
-            raise MisconfigurationException(
-                'Requested `accelerator="horovod"`, but Horovod is not installed.'
-                "Install with \n $HOROVOD_WITH_PYTORCH=1 pip install horovod[pytorch]"
-            )
+    @property
+    def is_distributed(self):
+        # Used for custom plugins.
+        # Custom plugins should implement is_distributed property.
```

Comment:
what still requires this and how can we eliminate this requirement?

Reply:
Yes, it used in Lite and Data_connector